src/grimoirelab/core/scheduler/scheduler.py

```diff
@@ -154,7 +155,16 @@ def _is_job_removed_or_stopped(job: Job, queue: str) -> bool:
     try:
         connection = django_rq.get_connection(queue)
         job_rq = rq.job.Job.fetch(job.uuid, connection=connection)
-        return job_rq.get_status() in RQ_JOB_STOPPED_STATUS
+        status = job_rq.get_status()
+        if status in RQ_JOB_STOPPED_STATUS:
+            return True
+        elif status == rq.job.JobStatus.STARTED:
+            # Sometimes, the worker may be forcibly stopped, leaving the job
+            # in the STARTED status. We need to check if the job has expired
+            # due to a missing heartbeat.
+            expiration_date = StartedJobRegistry(queue, connection).get_expiration_time(job_rq)
+            if expiration_date < datetime.datetime.now():
+                return True
```

Comment:
This piece of code should be rewritten this way or there will be cases when the function doesn't return anything.

```suggestion
        if status == rq.job.JobStatus.STARTED:
            # Sometimes, the worker may be forcibly stopped, leaving the job
            # in the STARTED status. We need to check if the job has expired
            # due to a missing heartbeat.
            expiration_date = StartedJobRegistry(queue, connection).get_expiration_time(job_rq)
            return expiration_date < datetime.datetime.now():
        else:
            return status in RQ_JOB_STOPPED_STATUS:

```